apps/finance/test/finance.js

```diff
@@ -636,28 +643,28 @@ contract('Finance App', ([root, owner, recipient]) => {
 
             it('can transition periods externally to remove deadlock for direct deposits', async () => {
                 const sentWei = 10
-                const prevVaultBalance = (await getBalance(vault.address)).toNumber()
+                const prevVaultBalance = parseInt(await getBalance(vault.address), 10)
 
                 await finance.tryTransitionAccountingPeriod(maxTransitions)
 
                 const receipt = await finance.send(sentWei, { gas: 3e5 })
                 const transactionId = getEventArgument(receipt, 'NewTransaction', 'transactionId')
-                const [periodId, amount, paymentId, paymentExecutionNumber, token, entity, incoming, date] = await finance.getTransaction(transactionId)
+                const data = await finance.getTransaction(transactionId)
 
-                assert.equal(amount, sentWei, 'app should have received ETH and sent it to vault')
-                assert.equal((await getBalance(vault.address)).valueOf(), prevVaultBalance + sentWei, 'app should have received ETH and sent it to vault')
+                assert.equal(data.amount, sentWei, 'app should have received ETH and sent it to vault')
+                assert.equal(parseInt(await getBalance(vault.address), 10), prevVaultBalance + sentWei, 'app should have received ETH and sent it to vault')
             })
 
             it('non-activity accounting periods have no transactions', async () => {
                 await finance.tryTransitionAccountingPeriod(5)
 
-                const [isCurrent, start, end, firstTx, lastTx] = await finance.getPeriod(2)
+                const data = await finance.getPeriod(2)
 
-                assert.isFalse(isCurrent, 'shouldnt be current period')
-                assert.equal(start.valueOf(), PERIOD_DURATION * 2 + 1, 'should have correct start date')
-                assert.equal(end.valueOf(), PERIOD_DURATION * 3, 'should have correct end date')
-                assert.equal(firstTx, 0, 'should have empty txs')
-                assert.equal(lastTx, 0, 'should have empty txs')
+                assert.isFalse(data.isCurrent, 'shouldnt be current period')
+                assert.equal(data.startTime.toNumber(), PERIOD_DURATION * 2 + 1, 'should have correct start date')
+                assert.equal(data.endTime.toNumber(), PERIOD_DURATION * 3, 'should have correct end date')
```

Comment:
Just in case, I wonder if we can use `.toString()` here instead. We should in general prefer to serialize BNs as strings; JS will transparently handle the cast from number to string for the other values.